docs/domains/Domain.json

```diff
@@ -235,6 +235,8 @@
           "enum": [
             "Progressing",
             "Available",
+            "OnlineUpdateComplete",
```

Comment:
I could be wrong on this, but putting the online update status in the domain condition I think (a) over complicates the domain condition, and (b) conflates two different concepts.  Perhaps online update should have its own condition with "Inapplicable", "Progressing", "Failed", and "Cancel" states or some-such, and the "Domain" status can simply remain at Progressing, Available, and Failed.  Regardless, if this condition is going to be updated, then its 'description' field needs to be updated.   

@rjeberhard Thoughts?

Reply:
On a related note, should we discuss adding corresponding Events for online updates?

Reply:
I put this link in the Slack thread, but also including here: https://dev.to/maelvls/what-the-heck-are-kubernetes-conditions-for-4je7. The design of Kubernetes conditions is supposed to be a "simpler" alternative to a state machine... You get simple Booleans telling you if the resource is in a given state, e.g. Running, Ready, etc. We can have as many or as few conditions as we want, but each condition has to have the value of "True", "False", or "Unknown".

We can and should also have other status fields. I like conditions because they are very simple to check.

Reply:
Looks good to me now, but just to confirm:
- Progressing, Available, Failed all continue to work intuitively (progressing if processing a domain resource change, available if change + introspect succeeded, failed if not)
- ConfigChangesPendingRestart can be true regardless of the value of other conditions
- ConfigChangesPendingRestart only transitions back to false once all pods no longer have the label (even if the operator restarted while the affected pods were still running).